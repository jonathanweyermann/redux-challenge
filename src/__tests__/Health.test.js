import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Health from '../health';
import { shallow, mount } from 'enzyme';
import '@testing-library/jest-dom';

describe('health component', () => {
  it('creates a low health snapshot', () => {
		const wrapper = mount(<Health value={42} displayValue={true} />);
	  expect(wrapper).toMatchSnapshot();
  });
  it('creates a medium health snapshot', () => {
		const wrapper = mount(<Health value={65} displayValue={true} />);
	  expect(wrapper).toMatchSnapshot();
  });
  it('creates a high health snapshot', () => {
		const wrapper = mount(<Health value={88} displayValue={true} />);
	  expect(wrapper).toMatchSnapshot();
  });
  it('creates a high health snapshot with no display number', () => {
		const wrapper = mount(<Health value={88} />);
	  expect(wrapper).toMatchSnapshot();
  });
});
