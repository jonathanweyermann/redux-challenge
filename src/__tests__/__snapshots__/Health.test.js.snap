// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`health component creates a high health snapshot 1`] = `
<Health
  displayValue={true}
  value={88}
>
  <div
    className="container"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <div
      className="health-value-display"
    >
      88
    </div>
    <div
      className="full-bar"
    >
      <div
        className="solid-bar"
        data-testid="solid-bar"
        style={
          Object {
            "backgroundColor": "#5cb85c",
            "width": "calc(100% * 0.88)",
          }
        }
      />
    </div>
  </div>
</Health>
`;

exports[`health component creates a high health snapshot with no display number 1`] = `
<Health
  value={88}
>
  <div
    className="container"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <div
      className="full-bar"
    >
      <div
        className="solid-bar"
        data-testid="solid-bar"
        style={
          Object {
            "backgroundColor": "#5cb85c",
            "width": "calc(100% * 0.88)",
          }
        }
      />
    </div>
  </div>
</Health>
`;

exports[`health component creates a low health snapshot 1`] = `
<Health
  displayValue={true}
  value={42}
>
  <div
    className="container"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <div
      className="health-value-display"
    >
      42
    </div>
    <div
      className="full-bar"
    >
      <div
        className="solid-bar"
        data-testid="solid-bar"
        style={
          Object {
            "backgroundColor": "#D9534F",
            "width": "calc(100% * 0.42)",
          }
        }
      />
    </div>
  </div>
</Health>
`;

exports[`health component creates a medium health snapshot 1`] = `
<Health
  displayValue={true}
  value={65}
>
  <div
    className="container"
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <div
      className="health-value-display"
    >
      65
    </div>
    <div
      className="full-bar"
    >
      <div
        className="solid-bar"
        data-testid="solid-bar"
        style={
          Object {
            "backgroundColor": "#f0ad4e",
            "width": "calc(100% * 0.65)",
          }
        }
      />
    </div>
  </div>
</Health>
`;
